src/d/actor/d_a_shutter.cpp

```diff
@@ -5,80 +5,292 @@
+            }
+            fVar4 = cLib_addCalc(&mcXyz[0].x, -m_move_len[mType] - m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            fVar5 = cLib_addCalc(&mcXyz[1].x, m_move_len[mType] + m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            if ((dStage_stagInfo_GetSaveTbl(dComIfGp_getStage().getStagInfo()) == 5) && (field_0x33A == 75)) {
+                fopAcM_seStart(this, JA_SE_OBJ_B_SHUTTER_STOP, 0);
+                dComIfGp_getVibration().StartShock(4, -0x21, cXyz(0.0f, 1.0f, 0.0f));
+            }
+            if (fVar4 != 0.0f){
+                return;
+            } 
+            if (fVar5 != 0.0f) {
+                return;
+            }
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+        case 3: //CLOSE
+        {
+            if (field_0x339 == 0) {
+                field_0x339 = 1;
+            }
+            fVar4 = cLib_addCalc(&mcXyz[0].x, -m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            fVar5 = cLib_addCalc(&mcXyz[1].x, m_width[mType] / 2.0f, 0.1f, maxVel, minVel);
+            if (fVar4 != 0.0f) {
+                return;
+            }
+            if (fVar5 != 0.0f) {
+                return;
+            }
+            dComIfGp_evmng_cutEnd(mStaffId);
+            return;
+        }
+    }
+    dComIfGp_evmng_cutEnd(mStaffId);
 }
 
 /* 00000B14-00000CF0       .text demo__11daShutter_cFv */
 void daShutter_c::demo() {
-    /* Nonmatching */
+    u8 bVar3 = fopAcM_isSwitch(this, mNo);
+    if (field_0x320 == 0) {
+        if ((bVar3 != (u8)mSwitch) && (mCount < 0)) {
+            if (!bVar3) {
+                field_0x320 = 2;
+            }
+            else {
+                field_0x320 = 1;
+            }
+        }
+    }
+    if (eventInfo.checkCommandDemoAccrpt()) {
+        if ((dComIfGp_evmng_startCheck(mOpenEventIdx)) && (field_0x320 == 1)) {
+            field_0x320 = 0;
+        }
+        if ((dComIfGp_evmng_startCheck(mCloseEventIdx)) && (field_0x320 == 2)) {
+            field_0x320 = 0;
+        }
+        if (dComIfGp_evmng_endCheck(mOpenEventIdx) || dComIfGp_evmng_endCheck(mCloseEventIdx)) {
+            dComIfGp_event_reset();
+        }
+        mStaffId = dComIfGp_evmng_getMyStaffId(m_staff_name[mType], NULL, 0);
+        shutter_move();
+    }
+    else if ((field_0x320 == 1) && (mOpenEventIdx != 0)) {
+        fopAcM_orderOtherEventId(this, mOpenEventIdx, 0xff, 0xffff, 0, 1);
+        eventInfo.mCondition = eventInfo.mCondition | 2;
+    }
+    else if ((field_0x320 == 2) && (mCloseEventIdx != 0)) {
+        fopAcM_orderOtherEventId(this, mCloseEventIdx, 0xff, 0xffff, 0, 1);
+        eventInfo.mCondition = eventInfo.mCondition | 2;
+    }
 }
 
 /* 00000CF0-00000DD8       .text _draw__11daShutter_cFv */
 bool daShutter_c::_draw() {
-    /* Nonmatching */
+    cXyz actorPos;
+    for (int i = 0; i < (int)ARRAY_SIZE(mMtx); i++) {
+        actorPos.x = current.pos.x;
+        actorPos.y = current.pos.y;
+        actorPos.z = current.pos.z;
```

Comment:
```suggestion
        actorPos = current.pos;
```
Might be the = inline? Ignore if you already tried that and it didn't match